### PR TITLE
fix(stages): Monorepo Primitive Refactoring

### LINKED ISSUE #48: fix(stages): Monorepo Management
**Description**

Path localization happens in an unsafe way inside individual stage handlers, many of which are redundant. This was an initial port to make space for this nice refactor as a followup.

This ticket is to pull all monorepo directory management and path localization into the `Monorepo` primitive in a safe way and have the `Monorepo` be propagated down to individual stages.

### PR DESCRIPTION
**Pull request by refcell**
**Description**

Fixes #48

This PR refactors Monorepo management into a primitive that can safely manage path localization.

### CODE REVIEW

**Comment by refcell** [comment]
Current dependencies on/for this PR:
* main
  * **PR #49** <a href="https://app.graphite.dev/github/pr/anton-rs/op-up/49" target="_blank">Graphite</a>  👈

This comment was auto-generated by [Graphite](https://app.graphite.dev/github/pr/anton-rs/op-up/49?utm_source=stack-comment).

File: crates/stages/src/stages/l1_genesis.rs [hunk 1]
```diff
@@ -25,55 +17,32 @@ impl crate::Stage for L1Genesis {
...
-            .as_ref()
-            .ok_or(eyre::eyre!("missing l1 genesis file"))?;
+        let deploy_config = self.monorepo.deploy_config();
+        let deploy_config = path_to_str!(deploy_config)?;
+        let allocs = self.monorepo.allocs();
+        let allocs = path_to_str!(allocs)?;
+        let l1_genesis = self.monorepo.l1_genesis();
+        let addresses_json = self.monorepo.addresses_json();
+        let addresses_json = path_to_str!(addresses_json)?;
+        let op_node_dir = self.monorepo.op_node_dir();
```
Comment (merklefruit) [inline crates/stages/src/stages/l1_genesis.rs:27]: lovely macro

**Review by merklefruit — APPROVED** [review]
stages just got a LOT simpler, lfg

**Comment by merklefruit** [comment]
after #50

**Pull request merged**